extensions/wikia/Flags/scripts/FlagsModal.js

```diff
@@ -94,9 +94,15 @@ require(
 				var template;
 
 				loader.processStyle(res.styles);
-				template = res.mustache[0];
 
-				flagsData[0].flags = prepareFlagsData(flagsData[0].flags);
+				if (flagsData[0].flags) {
+					template = res.mustache[0];
+					flagsData[0].flags = prepareFlagsData(flagsData[0].flags);
+				} else if (flagsData[0].emptyMessage) {
+					template = res.mustache[1];
+				} else if (flagsData[0].exceptionMessage) {
+					template = res.mustache[2];
+				}
```

Comment:
My suggestion would be to leave this in place, in this case splitting this off to a separate function this would be the only place that uses that code. Since the logic is very simple, I'd opt to leave it here -- if we end up adding more code we can revisit and potentially break this up as you suggested, Kamil.